pandas/core/strings/base.py

```diff
@@ -1,274 +0,0 @@
-from __future__ import annotations
-
-import abc
-from typing import (
-    TYPE_CHECKING,
-    Literal,
-    Self,
-)
-
-from pandas._libs import lib
-
-if TYPE_CHECKING:
-    from collections.abc import (
-        Callable,
-        Sequence,
-    )
-    import re
-
-    from pandas._typing import (
-        NpDtype,
-        Scalar,
-    )
-
-
-class BaseStringArrayMethods(abc.ABC):
```

Comment:
FWIW, would this mixin be good to expose publicly one day if an EA author wants to override e.g. `.str.contains`?